src/libraries/Common/src/Interop/BSD/System.Native/Interop.Sysctl.cs

```diff
@@ -75,10 +73,24 @@ private static unsafe void Sysctl(int* name, int name_len, ref byte* value, ref
                 {
                     Marshal.FreeHGlobal((IntPtr)value);
                 }
-                throw new InvalidOperationException(SR.Format(SR.InvalidSysctl, *name, Marshal.GetLastPInvokeError()));
+                ThrowInvalidSysctlException(name, name_len);
             }
 
-            len = (int)bytesLength;
+            len = (uint)bytesLength;
+        }
+
+        private static unsafe InvalidOperationException ThrowInvalidSysctlException(int* name, uint name_len)
+        {
+            int mib0 = name[0];
+            int mib1 = name_len > 1 ? name[1] : -1;
+            int mib2 = name_len > 2 ? name[2] : -1;
+            int mib3 = name_len > 3 ? name[3] : -1;
+            throw new InvalidOperationException(SR.Format(SR.InvalidSysctl,
+                                                           mib0,
+                                                           mib1,
+                                                           mib2,
+                                                           mib3,
+                                                           Marshal.GetLastPInvokeError()));
```

Comment:
The last PInvokeError may be overwritten between this Sysctl and this point. For example, `NativeMemory.Free` may overwrite it in some situations. It would be better to call `Marshal.GetLastPInvokeError()` right after `if (ret != 0)` to make sure that it won't get overwritten.